endpoints/books/controller.py

```diff
@@ -33,38 +36,49 @@ def post(self):
         try:
             current_user = get_jwt_identity()
             book = request.get_json()
-            book['user_id'] = current_user
+            existing_book = userBookRepository.findByName(book['name'])
+            if existing_book['name'] == book['name'] and existing_book['user_id'] == current_user['id']:
+                return {'message': 'Resource already exists', 'time': datetime.datetime.now().isoformat()}, 422
+            book['user_id'] = current_user['id']
             userBookRepository.add(book)
             return marshal(book, book_list_fields)
 
         except Exception as error:
-            return {'message': 'Something went wrong', 'timestamp': round(time.time())}, 500
-            
+            logging.error(f'{error}')
```

Comment:
You could try `logging.exception` as well.

Reply:
I'll test it tomorrow!